src/util/tev_curl.h

```diff
@@ -0,0 +1,111 @@
+/*
+   SSSD
+
+   libcurl tevent integration
+
+   Copyright (C) Red Hat, 2016
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __TEV_CURL_H
+#define __TEV_CURL_H
+
+#include <talloc.h>
+#include <tevent.h>
+
+#include "util/sss_iobuf.h"
+
+/**
+ * @brief Supported HTTP requests
+ */
+enum tcurl_http_request {
+    HTTP_GET,
+    HTTP_PUT,
+    HTTP_DELETE,
+};
```

Comment:
Those values conflicts with `http-parser` in `secrets` responder. Can you give them some prefix? `TCURL_HTTP_*`?